Thanks for the changes @oguzkocer! I'm always unsure about using Thread.sleep, especially because of the introduction of a possible precedent. Having said that I think this use case is +- valid and hopefully we won't start over-using it. Implementing retry for cases when the backend returns `unknown_plugin` might be another option, but implementing retry within FluxC is probably not a great practice either, so 🤷.

I run the test several times and unfortunately, it failed with the "UNKNOWN_SITE_PLUGIN" one time :(. So it seems that half a second might not be enough in some cases. Wdyt?

```
2021-04-14 06:15:15.856 29369-29421/? E/Volley: [2815] BasicNetwork.performRequest: Unexpected response code 404 for https://public-api.wordpress.com/rest/v1.2/sites/137437733/plugins/react%2Freact/?locale=en_US
2021-04-14 06:15:15.857 29369-29369/? E/WordPress-API: Volley error on https://public-api.wordpress.com/rest/v1.2/sites/137437733/plugins/react%2Freact/?locale=en_US
    com.android.volley.ClientError
        at com.android.volley.toolbox.BasicNetwork.performRequest(BasicNetwork.java:199)
        at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:131)
        at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:111)
        at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:90)
2021-04-14 06:15:15.861 29369-29443/? E/AndroidRuntime: FATAL EXCEPTION: pool-1-thread-3
    Process: org.wordpress.android.fluxc.example, PID: 29369
    org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
        at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:530)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:512)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:504)
        at org.greenrobot.eventbus.AsyncPoster.run(AsyncPoster.java:46)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: org.greenrobot.eventbus.EventBusException: Invoking subscriber failed
        at org.greenrobot.eventbus.EventBus.handleSubscriberException(EventBus.java:530)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:512)
        at org.greenrobot.eventbus.EventBus.postToSubscription(EventBus.java:433)
        at org.greenrobot.eventbus.EventBus.postSingleEventForEventType(EventBus.java:414)
        at org.greenrobot.eventbus.EventBus.postSingleEvent(EventBus.java:387)
        at org.greenrobot.eventbus.EventBus.post(EventBus.java:268)
        at org.wordpress.android.fluxc.Dispatcher.emitChange(Dispatcher.java:41)
        at org.wordpress.android.fluxc.store.Store.emitChange(Store.java:32)
        at org.wordpress.android.fluxc.store.PluginStore.configuredSitePlugin(PluginStore.java:755)
        at org.wordpress.android.fluxc.store.PluginStore.onAction(PluginStore.java:597)
        at java.lang.reflect.Method.invoke(Native Method)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:510)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:504) 
        at org.greenrobot.eventbus.AsyncPoster.run(AsyncPoster.java:46) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
     Caused by: java.lang.AssertionError: expected:<CONFIGURED_SITE_PLUGIN> but was:<UNKNOWN_SITE_PLUGIN>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:144)
        at org.wordpress.android.fluxc.release.ReleaseStack_PluginTestJetpack.onSitePluginConfigured(ReleaseStack_PluginTestJetpack.java:391)
        at java.lang.reflect.Method.invoke(Native Method)
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:510)
        at org.greenrobot.eventbus.EventBus.postToSubscription(EventBus.java:433) 
        at org.greenrobot.eventbus.EventBus.postSingleEventForEventType(EventBus.java:414) 
        at org.greenrobot.eventbus.EventBus.postSingleEvent(EventBus.java:387) 
        at org.greenrobot.eventbus.EventBus.post(EventBus.java:268) 
        at org.wordpress.android.fluxc.Dispatcher.emitChange(Dispatcher.java:41) 
        at org.wordpress.android.fluxc.store.Store.emitChange(Store.java:32) 
        at org.wordpress.android.fluxc.store.PluginStore.configuredSitePlugin(PluginStore.java:755) 
        at org.wordpress.android.fluxc.store.PluginStore.onAction(PluginStore.java:597) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:510) 
        at org.greenrobot.eventbus.EventBus.invokeSubscriber(EventBus.java:504) 
        at org.greenrobot.eventbus.AsyncPoster.run(AsyncPoster.java:46) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
```

--- a/fluxc/src/main/java/org/wordpress/android/fluxc/store/PluginStore.java
+++ b/fluxc/src/main/java/org/wordpress/android/fluxc/store/PluginStore.java
@@ -819,8 +819,8 @@ public class PluginStore extends Store {
         // Once the plugin is installed activate it and enable auto-updates
         if (!payload.isError() && payload.plugin != null) {
             try {
-                // Give half a second to the server as otherwise the following configure call may fail
-                Thread.sleep(500);
+                // Give a second to the server as otherwise the following configure call may fail
+                Thread.sleep(1000);
             } catch (InterruptedException e) {
                 // https://www.javaspecialists.eu/archive/Issue056-Shutting-down-Threads-Cleanly.html
                 Thread.currentThread().interrupt();

